internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -491,3 +505,182 @@ func (r PersistentVolumeClaimReconciler) supportsReclaimSpace(driverName string)
 
 	return false
 }
+
+func (r *PersistentVolumeClaimReconciler) findChildEncryptionKeyRotationCronJob(
+	ctx context.Context,
+	logger *logr.Logger,
+	req *reconcile.Request) (*csiaddonsv1alpha1.EncryptionKeyRotationCronJob, error) {
+	var childJobs csiaddonsv1alpha1.EncryptionKeyRotationCronJobList
+	var activeJob *csiaddonsv1alpha1.EncryptionKeyRotationCronJob
+
+	err := r.List(ctx, &childJobs, client.InNamespace(req.Namespace), client.MatchingFields{jobOwnerKey: req.Name})
+	if err != nil {
+		logger.Error(err, "failed to list child encryptionkeyrotationcronjobs")
+		return activeJob, fmt.Errorf("failed to list encryptionkeyrotationcronjobs: %v", err)
+	}
+
+	for i, job := range childJobs.Items {
+		// max allowed job is 1
+		if i == 0 {
+			activeJob = &job
+			continue
+		}
+
+		// delete the rest if found
+		if err = r.Delete(ctx, &job); err != nil {
+			if client.IgnoreNotFound(err) != nil {
+				logger.Error(err, "failed to delete extraneous child encryptionkeyrotationcronjob", "EncryptionKeyrotationCronJobName", &job.Name)
+				return nil, fmt.Errorf("failed to delete extraneous child encryptionkeyrotationcronjob: %w", err)
+			}
+		}
+	}
+
+	return activeJob, nil
+}
+
+func (r *PersistentVolumeClaimReconciler) determineEncryptionKeyRotationSchedule(
```

Comment:
lets not have duplicate helper use existing on and pass the required key as input

Reply:
The logic differs between the existing function and this one. We do not check namespaces and pvc annotations but only the storage class.

Reply:
why, if we just refactor we can support all the functionalities what we have for reclaimspace, any concern with that?

Reply:
Did not know we could do that (as it differed from the design). In that case, we can make key rotation work with namespace annotations as well. Win win!